File: csharp/test/Ice/inheritance/ICI.cs
// Copyright (c) ZeroC, Inc.

namespace Ice.inheritance;

public sealed class ICI : Test.MA.ICDisp_
{
    public ICI()
    {
    }

    public override Test.MA.IAPrx iaop(Test.MA.IAPrx p, Ice.Current current) => p;

    public override Test.MA.ICPrx icop(Test.MA.ICPrx p, Ice.Current current) => p;

    public override Test.MB.IB1Prx ib1op(Test.MB.IB1Prx p, Ice.Current current) => p;

    public override Test.MB.IB2Prx ib2op(Test.MB.IB2Prx p, Ice.Current current) => p;
}

File: java/test/Ice/inheritance/ICI.java
// **********************************************************************
//
// Copyright (c) 2002
// ZeroC, Inc.
// Billerica, MA, USA
//
// All Rights Reserved.
//
// Ice is free software; you can redistribute it and/or modify it under
// the terms of the GNU General Public License version 2 as published by
// the Free Software Foundation.
//
// **********************************************************************

public final class ICI extends MA._ICDisp
{
    public
    ICI()
    {
    }

    public MA.IAPrx
    iaop(MA.IAPrx p, Ice.Current current)
    {
        return p;
    }

    public MA.ICPrx
    icop(MA.ICPrx p, Ice.Current current)
    {
        return p;
    }

    public MB.IB1Prx
    ib1op(MB.IB1Prx p, Ice.Current current)
    {
        return p;
    }

    public MB.IB2Prx
    ib2op(MB.IB2Prx p, Ice.Current current)
    {
        return p;
    }
}
